sopel/irc.py

```diff
@@ -196,7 +196,8 @@ def restart(self, message):
 
     def quit(self, message):
         """Disconnect from IRC and close the bot."""
-        self.write(['QUIT'], message)
+        if self.connected:  # Only send QUIT message if socket is open
```

Comment:
`connected` is only set if the socket has been created for the `async_chat` object. This does not require a successful IRC connection, just an open connection to the server.